### ISSUE TITLE
全新数据集的训练

### ISSUE THREAD

**Issue by jjm666if** [zhiyuanyou/DeQA-Score#9]
请教一下，当我使用全新的数据集，一万多张图像，mos值在0-5之间，使用lora微调之后，测出的plcc和srocc的指标只有0.4，我应该如何训练去提升他的指标

**Comment by zhiyuanyou** [comment]
norm有点错误，mos值应该在1-5之间。

**Comment by zhiyuanyou** [comment]
另外，注意norm mos的时候，std也需要跟着改变。

**Comment by jjm666if** [comment]
不好意思，写错了，是mos是1-5之间

**Comment by zhiyuanyou** [comment]
数据集是如何标注的？

如果不是单张score求mean标注的mos（比如，pair-wise标注），不适合根据高斯分布做soft label，可以尝试二值化的soft label。在数据合成的config中将"thre_std"和"thre_diff"都修改为一个很大的值，例如10000，保证全部数据都使用二值化的soft label。

另外，也可以考虑将dataset_type改成single，取消ranking loss，看看效果。

总之，使用ablation找到原因，之后针对性调整。

**Comment by jjm666if** [comment]
这是我数据的组织形式，单条 ，您帮忙看看有什么问题吗？
{
        "id": "EPICNN_Bikes-1_3.png->3.9565",
        "image": "../win5/images\\EPICNN_Bikes-1_3.png",
        "gt_score": 3.9565,
        "gt_score_norm": 4.238048299655003,
        "level_probs_org": [
            0.005297417832179546,
            1.0121564155643834,
            3.784589315480407e-07,
            2.7693484850598844e-22,
            3.965739501555572e-46
        ],
        "level_probs": [
            0.23804830489403694,
            0.7619516951059631,
            0,
            0,
            0
        ],
        "std": 0.2039,
        "std_norm": 0.22331745249438698,
        "conversations": [
            {
                "from": "human",
                "value": "Could you evaluate the quality of this image?\n<|image|>"
            },
            {
                "from": "gpt",
                "value": "The quality of the image is excellent."
            }
        ]
    },

**Comment by zhiyuanyou** [comment]
跟大多数数据集比起来，你这个std太小了。为了让训练更稳定，开源的代码里是fix了pred std是1。可能跟你的std相差太多了。可以试试把这个参数改一下：将 --use_fix_std True 改成  --use_fix_std False。

**Comment by jjm666if** [comment]
好的好的，非常感谢

**Comment by zhiyuanyou** [comment]
你好，你的数据集的std非常小，十分类似于TID2013。我有点怀疑你的数据不符合高斯分布的假设。

详见UNIQUE (Uncertainty-Aware Blind Image Quality Assessment in the Laboratory and Wild) 这篇paper：

> We excluded TID2013 [47] in our experiments because the MOS is computed by the number of winning times in a maximum of nine paired comparisons without suitable psychometric scaling [48], and does not satisfy the Gaussian assumption.

**Comment by jjm666if** [comment]
所以这个std的值对于训练过程非常重要是吗？另外请问你们是否有合并lora权重模型的代码，我写了一个不知道对不对。
import os
import torch
from peft import PeftModel
from transformers import AutoTokenizer
from src.model import MPLUGOwl2LlamaForCausalLM

def merge_lora_weights(
    base_model_path,
    checkpoint_path,
    output_path,  
    device="cuda" if torch.cuda.is_available() else "cpu",
):
  
    os.makedirs(output_path, exist_ok=True)
    
    # 加载基础模型
    base_model = MPLUGOwl2LlamaForCausalLM.from_pretrained(
        base_model_path,
        torch_dtype=torch.float16,
        device_map=device
    )

    # 加载并合并LoRA
    lora_model = PeftModel.from_pretrained(base_model, checkpoint_path)
    merged_model = lora_model.merge_and_unload()

    # 强制保存到指定路径
    merged_model.save_pretrained(output_path, safe_serialization=True)
    
    # 保存分词器
    tokenizer = AutoTokenizer.from_pretrained(base_model_path)
    tokenizer.save_pretrained(output_path)

    print(f"模型已保存到: {os.path.abspath(output_path)}")

if __name__ == "__main__":
  
    _BASE_MODEL = "/070025/DeQA-Score-main/ModelZoo/model" 
    _LORA_MODEL = "/070025/DeQA-Score-main/checkpoints/win5_nbu_3-28-20"
    _OUTPUT_DIR = "/070025/DeQA-Score-main/ModelZoo/model1"  # 改成您需要的任意路径
    
    merge_lora_weights(
        base_model_path=_BASE_MODEL,
        checkpoint_path=_LORA_MODEL,
        output_path=_OUTPUT_DIR  # 输出路径最终由这个参数决定
    )

**Comment by zhiyuanyou** [comment]
不是std重要，是score分布近似满足高斯分布重要。如果不近似满足高斯分布，我们的soft label就不能使用了。

**Comment by jjm666if** [comment]
了解了

**Comment by CharlesChen24** [comment]
有限资源，如果我的数据集，一张图只标一次，分值 [1,2,3,4,5] 中的一个。3的数量多于2,4数量多于1,5的数量，可以使用你这种模型训练吗？
那gt_score，gt_score_norm，std，std_norm等一些数据集的参数要怎么设置？
